src/liquidity_mining/liquidity_mining.rs

```diff
@@ -133,11 +136,20 @@ pub fn calculate_global_farm_rewards<Period: num_traits::CheckedSub + TryInto<u3
         .checked_mul_int(total_shares_z)
         .ok_or(MathError::Overflow)?;
 
-    let reward_per_period = yield_per_period
-        .checked_mul(&FixedU128::from(total_shares_z_adjusted))
+    let periods = TryInto::<u128>::try_into(periods_since_last_update).map_err(|_e| MathError::Overflow)?;
+
+    //(total_shares_z_adjusted * yield_per_period.into_inner() * periods)/FixedU128::DIV;
+    let calculated_rewards = to_u256!(total_shares_z_adjusted)
+        .checked_mul(yield_per_period.into_inner().into())
         .ok_or(MathError::Overflow)?
-        .min(FixedU128::from(max_reward_per_period));
+        .checked_mul(periods.into())
```

Comment:
Wouldnt you have the same issue of overflowing here at some point ?

it seems that it does:
(u128  * u128 * u128 ) / u128

you confident that total shares * yield_per_period * periods cannot overflow again ?

Reply:
doesn't into() convert these values to u256?

Reply:
2^128*3 > 2^256

Reply:
i would 1st calculate reward per period, then multiply that by number of periods

Reply:
I added test with really big numbers here: https://github.com/galacticcouncil/HydraDX-math/blob/81dc370de55dbfd02e02ffea94da99f2a6d250eb/src/liquidity_mining/tests.rs#L918-L923
`yield_per_period` can be max 1 but it's per period so usually it's smaller. e.g current `0.000_000_152_207_001_522` and `periods` is diff between 2 block numbers so it's less then u64.

Reply:
you can't do it because you loose precision

Reply:
`total_shares_z` - 128b, 
`yield_per_period` is `Perquintill` max value: 1_000_000_000_000_000_000, so no more than 64b should be used
`periods` - we are using u32 for block numbers
so I think U256 should be enough